### ISSUE TITLE
Filter function only applies if user logged in

### ISSUE THREAD

**Issue by m-hope** [AtlasOfLivingAustralia/profile-hub#574]
If a user is not logged in (ie: using a public collection), the filter function will allow them to set a filter list, but it won't apply to the collection.

Similarly, if a logged in user sets a filter, it will apply and restrict the profiles displayed, but if the user logs out the filter is ignored and all profiles are visible in Browse.  If the user logs back in then the filter will then apply again.

**Comment by m-hope** [comment]
There is currently a message on the filter page that appears when not logged in that states that the filter will only apply for this session... this is incorrect as the filter doesn't apply at all...

Either the filter function needs to be fixed so that non-logged in users can use it (ie: so the message is correct), or the message needs to be changed to indicate that you need to be logged in for the filter function to work.

If sprint time is limited, the last option is acceptable and preferred.

**Comment by Rita-C** [comment]
https://upsource.ala.org.au/profiles/review/PRFLS-CR-196

**Comment by RobinaSanderson** [comment]
Tested in profiles-dev environment using the collection ALA Test 2.

Steps:
1. ensure not logged in
2. open browse and note number of profiles in acacia = 102
3. apply the filter "IBRA Arnhem Coast - Flora" 
4. open browse and check number of profiles in acacia = 14

Filter is being applied.

**Comment by m-hope** [comment]
Confirmed.

**Comment by elywallis** [comment]
Tested in aws-profiles.ala.org.au using the only collection available when not logged on: Flora of Australia
Steps:
1. ensure not logged on
2. open browse and note number of profiles in Acacia = 1232
3. open Filter tab. Warning message states "You are not logged in. Any filter set will not persist beyond this browser session."
4. Apply filter "IBRA Arnhem Coast - Flora"
5. Re open Browse and check the number of profiles in Acacia = 45
Filter is being applied.
6. Close browser session. Open a new browser session immediately.
**Filter continues to be applied.

I have tested this on Chrome and IE but in both cases my organisation automatically logs me in via the browser, so the continued presence of the filter across browser sessions might be due to this? When I tried in Chrome in Incognito mode and the filter behaved as expected when not logged into ALA.
